server.js

```diff
@@ -1,25 +1,49 @@
 const express = require('express');
-const webpackDevMiddleware = require('webpack-dev-middleware');
 const webpack = require('webpack');
-const webpackConfig = require('./webpack.config.js');
 const app = express();
+const path = require('path');
+require('dotenv').config();
 
-const compiler = webpack(webpackConfig);
+if (process.env.NODE_ENV === 'development') {
+	const webpackDevMiddleware = require('webpack-dev-middleware');
+	const webpackHotMiddleware = require('webpack-hot-middleware');
+	const webpackConfig = require('./webpack.dev.js');
+	const compiler = webpack(webpackConfig);
+	app.use(
+		webpackDevMiddleware(compiler, {
+			hot: true,
+			publicPath: webpackConfig.output.publicPath
+		})
+	);
+	app.use(webpackHotMiddleware(compiler));
+    app.use(express.static(path.join(__dirname, '/')));
+    
+    app.get('/api', (req, res, next) => {
+		console.log('Should server mock data');
+		res.send('End Request....');
+    });
+    
+	app.use('*', function(req, res, next) {
+		let filename = path.join(compiler.outputPath, 'index.html');
+		compiler.outputFileSystem.readFile(filename, function(err, result) {
```

Comment:
@kimobrian please explain what happens here.:pray:

Reply:
![image](https://user-images.githubusercontent.com/22055697/36384532-a7abd05a-15a0-11e8-97c3-405de406767c.png)
 @danielwangai When we build in production, this is the output. It's the same output in dev only that dev holds it in memory and holds it in the output space(outputFileSystem) so here I am just reading the file `dist/index.html` since that's what we expect from our config. I then try to read a file with the same name from the outputFileSystem and server it.

Reply:
Makes sense.